### DISCUSSION TITLE
Related to Code generation

### DISCUSSION THREAD

**Discussion by opsingh861** [Azure/data-api-builder#2393] [Q&A]
I am working on a project in which I want to generate code for the API and it will be directly pushed to VCS. DAB is working like a charm. But the only problem is that how I will include business logic in same code. So if some how I can get the code on which we are able to communicating to DB then in that I can integrate my business logic. Or suggest me if there is any better way using which I can integrate business logic here.
Thanks.

**Comment by seantleonard** [comment]
Hi @opsingh861, can you help me further understand what you're asking? 

> But the only problem is that how I will include business logic in same code. 
> how I can get the code on which we are able to communicating to DB then in that I can integrate my business logic. 

It seems like you're asking if you can have your own business logic run after DAB receives a request and before DAB sends the request to the database? e.g. your business logic performs some additional validation on request payload? If so, DAB doesn't support that scenario at this time. 

Please take a look at https://github.com/Azure/data-api-builder/issues/2363 to see if the proposal for additional field input validation aligns with what you are asking and provide feedback.

**Reply by opsingh861** [reply]
Thank you for your prompt response, @seantleonard . I'm curious about DAB's code generation capabilities. Specifically, I'm interested in knowing if we can access the generated code that defines routes and other functionalities. While we could manually write this code, DAB's approach seems to streamline the process by providing a direct endpoint.

Additionally, I'd like to understand how to introduce custom routes. For example, if DAB generates a /book route based on a schema, can we add a new route like /multiplyby2 to the same backend, logic for this route will be defined by us but I just want to integrate it in the existing server? This would allow us to perform additional operations without needing to run a separate server.

If we can access the generated code, it should be easier to integrate custom routes. I'm eager to learn more about this process.

**Reply by jmlane** [reply]
I would also be interested in an engine option for exporting the generated code to source files.

**Comment by JerryNixon** [comment]
This would be neat but is outside of any plans we have.
